modules/oe_migration_workflow/src/Plugin/migrate/process/SetWorkflowState.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration_workflow\Plugin\migrate\process;
+
+use Drupal\Core\Config\ConfigManagerInterface;
+use Drupal\Core\Config\Entity\ConfigEntityStorage;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+use Drupal\oe_migration\ValidConfigurableMigrationPluginInterface;
+use Drupal\workflows\WorkflowInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Sets the workflow state of an entity.
+ *
+ * If the input value is a valid workflow state, it will be returned
+ * untransformed. Else, if the workflow state is not set or is not a valid one,
+ * it will be set according to the status property, if the status is set to "1",
+ * then it will be set to the given "published_state", in any other case, it
+ * will be set to the given "unpublished_state".
+ *
+ * Compared to a static mapping, it can also handle cases where the source
+ * entity might not have a moderation state defined (NULL).
+ *
+ * Available configuration keys:
+ *   - workflow_config_name (mandatory): The destination workflow's
+ *     configuration name (string).
+ *     Default value: "workflows.workflow.oe_corporate_workflow".
+ *   - published_state (mandatory): The destination workflow's published state
+ *     (string).
+ *     Default value: "published".
+ *   - unpublished_state (mandatory): The destination workflow's unpublished
+ *     state (string).
+ *     Default value: "draft".
+ *
+ * @codingStandardsIgnoreStart
+ * Example with default configuration:
+ * @code
+ * process:
+ *   moderation_state:
+ *    - plugin: oe_migration_set_workflow_state
+ *      source: moderation_state
+ * @endcode
+ *
+ * Example with user defined configuration:
+ * @code
+ * process:
+ *   moderation_state:
+ *    - plugin: oe_migration_set_workflow_state
+ *      source: moderation_state
+ *      workflow_config_name: workflows.workflow.oe_corporate_workflow
+ *      published_state: published
+ *      unpublished_state: draft
+ * @endcode
+ * @codingStandardsIgnoreEnd
+ *
+ * @MigrateProcessPlugin(
+ *   id = "oe_migration_set_workflow_state"
+ * )
+ */
+class SetWorkflowState extends ProcessPluginBase implements ContainerFactoryPluginInterface, ValidConfigurableMigrationPluginInterface {
+
+  /**
+   * The configuration manager.
+   *
+   * @var \Drupal\Core\Config\ConfigManagerInterface
+   */
+  protected $configManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
```

Comment:
done.

Reply:
It's not an inherited exception but introduced by our implementation. By removing it, we are leaving the user with an undeclared exception. Since the override also introduces a new parameter, I would like to propose to remove the `{@inheritdoc}` and provide a full description of the constructor, as some core process plugins (Download, EntityExists, FileCopy, FileProcessBase, MachineName, MenuLinkParent, MigrationLookup and SubProcess) do when they override the constructor. Sorry, my omission.

Reply:
No, thats not the point. We do not put @throws in the docblock according to our coding standards. Anywhere. 

RE {@inheritdoc}, you can leave it. Thats not an issue if you don't capture all params.